File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/fixed-asset/assets-management/insurance-add-edit/fixed-asset-insurance.model.ts
import { FormGroup, FormBuilder, Validators } from "@angular/forms";
import * as moment from "moment";

export class FixedAssetInsuranceModel {
  public AssetInsurannceId: number = 0;
  public FixedAssetStockId: number = 0;
  public PolicyNumber: string;
  public Insurer: string;
  public InsuredValue: string;
  public InsuranceStartDate: string = moment().format("YYYY-MM-DD");
  public InsuranceEndDate: string = moment().format("YYYY-MM-DD");
  // public InsuranceEndDate: string = moment(new Date).format("dd/MM/yyyy");
  public ComprehensiveInsurance: string;
  public CreatedBy: number;
  public CreatedOn: string;
  public ModefiedBy: number;
  public ModefiedOn: string;

  public InsuranceValidators: FormGroup = null;

  constructor() {
    var _formBuilder = new FormBuilder();
    this.InsuranceValidators = _formBuilder.group({
      'PolicyNumber': ['', Validators.compose([Validators.required])],
      'Insurer': ['', Validators.compose([Validators.required])],
      'InsuredValue': ['', Validators.compose([Validators.required])],
      'InsuranceStartDate': ['', Validators.compose([Validators.required])],
      'InsuranceEndDate': ['', Validators.compose([Validators.required])],
      'ComprehensiveInsurance': ['', Validators.compose([Validators.required])],

    });
  }

  public IsDirty(fieldName): boolean {
    if (fieldName == undefined)
      return this.InsuranceValidators.dirty;
    else
      return this.InsuranceValidators.controls[fieldName].dirty;
  }

  public IsValid(): boolean {
    if (this.InsuranceValidators.valid) { return true; } else { return false; }
  }
  public IsValidCheck(fieldName, validator): boolean {
    if (fieldName == undefined) {
      return this.InsuranceValidators.valid;
    }
    else {
      return !(this.InsuranceValidators.hasError(validator, fieldName));
    }
  }

}

File: Code/Components/DanpheEMR.ServerModel/FixedAssetModels/FixedAssetInsuranceModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class FixedAssetInsuranceModel
    {
        [Key]
        public int AssetInsurannceId { get; set; }
        public int FixedAssetStockId { get; set; }
        public string PolicyNumber { get; set; }
        public string Insurer { get; set; }
        public string InsuredValue { get; set; }
        public DateTime? InsuranceStartDate { get; set; }
        public DateTime? InsuranceEndDate { get; set; }
        public string ComprehensiveInsurance { get; set; }
        public DateTime? CreatedOn { get; set; }
        public int? CreatedBy { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public int? ModifiedBy { get; set; }
    }
}
